projects/centeva-core/test/app/components/table.component.spec.ts

```diff
@@ -161,6 +163,44 @@ describe('Table Component tests', () => {
     expect(component.emitSearchChanged).toHaveBeenCalled();
   });
 
+  it('checkbox master toggle', () => {
+    const column = component.displayedColumns[5];
+
+    component.checkboxMasterToggle(column.Property);
+    
+    expect(component.emitCheckbox).toHaveBeenCalled();
+
+    let checkboxData = component.checkboxModels[column.Property]
+    expect(checkboxData.SelectionModel.selected.length).toBe(dataSource.Records.length);
+    expect(checkboxData.AllItemsSelected).toBe(true);
+    expect(checkboxData.AnyItemSelected).toBe(true);
+  });
+
+  it('checkbox toggle', () => {
```

Comment:
Might be worth having a test that makes sure having two checkbox columns in a table works as expected.